reward/staking_manager.go

```diff
@@ -78,14 +78,14 @@ func (sm *stakingManager) updateStakingCache(blockNum uint64) (*StakingInfo, err
 }
 
 // Subscribe setups a channel to listen chain head event and starts a goroutine to update staking cache.
-func (sm *stakingManager) subscribe() {
+func (sm *StakingManager) Subscribe() {
 	sm.chainHeadSub = sm.bc.SubscribeChainHeadEvent(sm.chainHeadCh)
 
 	go sm.handleChainHeadEvent()
 }
 
-func (sm *stakingManager) handleChainHeadEvent() {
-	defer sm.chainHeadSub.Unsubscribe()
+func (sm *StakingManager) handleChainHeadEvent() {
```

Comment:
Method description comment is missing.

Reply:
I'll make a new pr for method description comments